_posts/2016/2016-02-16-pipeline-elk-test-evaluation-2.md

```diff
@@ -0,0 +1,290 @@
+---
+layout: post
+title: "Automated test evaluation with ELK in the pipeline: Implemented solution"
+date: "2016-02-16 06:16:02"
+icon: wrench
+categories: tech-stories
+authors: ["Benjamin N.", "Bastian K."]
+---
+
+We implemented a Selenium test report database with Elasticsearch, Logstash, Docker, CircleCI and Jenkins to ease the test evaluation process in our Continuous Delivery Pipeline. Last week we have already introduced you to the [background of the project](https://developer.epages.com/blog/2016/02/11/pipeline-elk-test-evaluation-1.html) and today we want to get your hands on the actual development steps. Hence, this post showcases the various parts of the implemented solution and discusses the pragmatic benefits for our pipeline and our speed-up for massive regression test evaluation.
+
+Furthermore, this article should serve as an outline of the consolidated technical expertise gained throughout the engineering process of this project.
+
+## Implemented Solution
+
+To get the big picture for splitting the Scrum epic into several stories with tasks and acceptance criteria we created a visualization, which could distinctly highlight the various parts that needed to be implemented. The first draft of the blueprint was sketched by hand and looked similar to this:
+
+{% image blog-pipeline-elk-test-evaluation-blueprint.png %} The blueprint of the solution architecture {% endimage %}
+
+As you can see above, several components of our infrastructure will be affected and also involved throughout the development of this project. The middle tier shows the essential interdigitation of the underlying job chain in our pipeline. Usually a CDP run involves several prepare jobs; then a huge amount of install and patch jobs are run in parallel on the various VMs of the vCenter (top tier); afterwards a fingerprint of all machines is created and finally the ESF testsuite (and others) are run onto all vCenter VMs. Sometimes the testsuite is even running against an ePages VM before, during and after patching has started (zero-down-time tests), so don't take the blueprint to literally.
+
+After the tests have been run, the JSON logs should have been created inside every single Jenkins job. As of now the tricky implementation of this project starts. We have decided to split the implementation in 5 parts and the next sections will explain each one step-by-step.
+
+### Part 1 - Define the Test Object and Extend the Test Suite Reporter
```

Comment:
Define the Test Object and Extend the Test Suite Reporter > Define the test object and extend the test suite reporter